api/v1alpha1/clusterpackage_types.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//nolint:dupl
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:resource:scope=Cluster,shortName=clpkg
+//+kubebuilder:printcolumn:name=Desired version,type=string,JSONPath=".spec.packageInfo.version"
+//+kubebuilder:printcolumn:name=Installed version,type=string,JSONPath=".status.version"
+
+// ClusterPackage is the Schema for the clusterpackages API
+type ClusterPackage struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   PackageSpec   `json:"spec,omitempty"`
+	Status PackageStatus `json:"status,omitempty"`
+}
+
+func (pkg *ClusterPackage) GetSpec() *PackageSpec {
+	return &pkg.Spec
```

Comment:
Hm, I'm not sure TBH… Assuming with nil check you mean something like this:
```go
if pkg != nil {
  return &pkg.Spec
}
return nil
```

I think it would only move the nil pointer error to a later point, since we currently don't expect GetSpec() to return nil anywhere. With that in mind, I think it would be easier to maintain if we consider any point where a `ctrlpkg.Package` is nil as a bug rather than updating all usages of all methods that rely on it. What do you think?

Reply:
Yes, we can do it like that too :+1: